app/src/main/java/bg/zahov/app/ui/exercise/ExercisesScreen.kt

```diff
@@ -45,10 +45,17 @@ import bg.zahov.fitness.app.R
 
 @Composable
 fun ExercisesScreen(
+    args: String?,
     viewModel: ExerciseViewModel = viewModel(),
     navigateInfo: () -> Unit,
-    navigateBack: () -> Unit
+    navigateBack: () -> Unit,
 ) {
+    args?.let {
```

Comment:
you can use saveStateHandle directly in the view model  -https://developer.android.com/topic/libraries/architecture/viewmodel/viewmodel-savedstate#savedstate-compose-state

Reply:
👍